docs/docs/workflows/terraform-module.md

```diff
@@ -0,0 +1,84 @@
+This allows to upgrade the cluster to a newer Constellation version by simply updating the module source.
+
+In all examples remember to replace `$VERSION`.
+
+### AWS example
+```
+module "aws-constellation" {
+  source = "https://github.com/edgelesssys/constellation/releases/download/$VERSION/terraform-module.zip//modules/aws-constellation" # TODO check the link (name the to be zipped dir modules) during the release. So far only tested with https://tfmodule.s3.eu-west-1.amazonaws.com/modulesV2.zip//modules/aws-constellation
+  name        = "aws-example"
+  image       = "$VERSION"
+  zone        = "us-east-2c"
+  name_prefix = "example"
+  node_groups = {
+    control_plane_default = {
+      role          = "control-plane"
+      zone          = "us-east-2c"
+      instance_type = "m6a.xlarge"
+      disk_size     = 30
+      disk_type     = "gp3"
+      initial_count = 2
+    },
+    worker_default = {
+      role          = "worker"
+      zone          = "us-east-2c"
+      instance_type = "m6a.xlarge"
+      disk_size     = 30
+      disk_type     = "gp3"
+      initial_count = 2
+    }
+  }
+}
+```
+
+### GCP example
+```
+module "aws-constellation" {
+  source = "https://github.com/edgelesssys/constellation/releases/download/$VERSION/terraform-module.zip" # TODO check the link during the release. So far only tested with https://tfmodule.s3.eu-west-1.amazonaws.com/modulesV2.zip//modules/aws-constellation
+  name    = "gcp-ex"
+  project = "constellation-331613"
+  service_account_id = "ex"
+  image   = "$VERSION"
+  zone    = "europe-west2-a"
+  node_groups = {
+    control_plane_default = {
+      role          = "control-plane"
+      zone          = "europe-west2-a"
+      instance_type = "n2d-standard-4"
+      disk_size     = 30
+      disk_type     = "pd-ssd"
+      initial_count = 2
+    },
+    worker_default = {
+      role          = "worker"
+      zone          = "europe-west2-a"
+      instance_type = "n2d-standard-4"
+      disk_size     = 30
+      disk_type     = "pd-ssd"
+      initial_count = 2
+    }
+  }
+}
+```
+
+## Custom setup
+If you need to separate IAM and cluster management or need custom infrastructure, you can also call the submodules individually.
+You can have a look at the respective convenience module (`{csp}-constellation`) for the structure.
+The submodules are:
+- `constellation-cluster`: used to fetch the CLI, create the Constellation configuration and to apply the cluster
+- `fetch-image`: used to translate the Constellation image version to the image ID of the cloud service provider
+- `infrastructure/{csp}`: contains the cluster infrastructure resources
+- `infrastructure/iam/{csp}`: contains the resources to set up IAM
+
+## Cluster upgrades
+using the remote source:
+1. update the module source
+2. update the image / Kubernetes version (if explicitly set)
+3. `terraform init --upgrade`
+4. `terraform apply`
```

Comment:
This looks like a draft that needs to be elaborated. Look at "First steps" in "Getting started" for an example of how instructions should be written. Also see https://learn.microsoft.com/en-us/style-guide/procedures-instructions/writing-step-by-step-instructions